src/services/DfusionService.ts

```diff
@@ -156,10 +167,40 @@ export class DfusionRepoImpl implements DfusionService {
   }
 
   public watchOrderPlacement(params: WatchOrderPlacementParams) {
-    this._contract.events
-      .OrderPlacement()
+    const OrderPlacement = this._contract.events.OrderPlacement
+    const subscriptions: Map<string, ContractEventEmitter<OrderPlacement>> = new Map()
+    if (this._tokenIdsFilter) {
+      const tokenListDescription = this._tokenIdsFilter.join(', ')
+      subscriptions.set(
+        'Orders whose Buy Token is ' + tokenListDescription,
+        OrderPlacement({
+          filter: { buyToken: this._tokenIdsFilter },
+        }),
+      )
+
+      subscriptions.set(
+        'Orders whose Sell Token is ' + tokenListDescription,
+        OrderPlacement({
+          filter: { sellToken: this._tokenIdsFilter },
+        }),
+      )
```

Comment:
I thought it should be possible to essentially produce this kind of [A || B topic](https://github.com/ethereumproject/wiki/wiki/JSON-RPC#parameters-58)

And it is, try `await contract.getPastEvents('OrderPlacement', {fromBlock: 10327950, filter: {sellToken:[1,2], buyToken:[3,4]}})` and look at all sell/buy permutations.
It matches everything with everything
```
topics = OrderPlacement && SELL(1 || 2) && BUY(3 || 4)
```



So one subscription of 
```js
OrderPlacement({
          filter: { buyToken: this._tokenIdsFilter, sellToken: this._tokenIdsFilter },
        }),
```
is enough :raised_hands:

Reply:
```suggestion
    if (this._tokenIdsFilter) {
      const tokenListDescription = this._tokenIdsFilter.join(', ')
      subscriptions.set(
        'Orders whose Buy or Sell Token is ' + tokenListDescription,
        OrderPlacement({
          filter: { buyToken: this._tokenIdsFilter, sellToken: this._tokenIdsFilter },
        }),
      )
```

like this

Reply:
Thanks for this Dima! I'll improve it in another PR. I'll share it already. Promise I'm doing this the moment I get a minute for it :)

Reply:
I'll leave it open, and create an issue!